src/Ens.ts

```diff
@@ -228,20 +230,8 @@ export default class Ens extends NamingService {
     });
   }
 
-  async location(domain: string): Promise<DomainLocation> {
-    const [registry, resolver, owner] = await Promise.all([
-      this.registryAddress(domain),
-      this.resolver(domain),
-      this.owner(domain),
-    ]);
-
-    return {
-      registry,
-      resolver,
-      networkId: this.network,
-      blockchain: BlockchainType.ETH,
-      owner,
-    };
+  locations(domains: string[]): Promise<Locations> {
+    return GetLocations(domains, this);
```

Comment:
@DancingAxolotl I know it's a draft, but i would avoid splitting the logic like that (passing this instance into utility method). It's better to implement something inside Resolution class as private method.